docs/rules/no-reject-literal.md

```diff
@@ -0,0 +1,57 @@
+# disallow using literals as Promise rejection reasons (no-reject-literal)
+
+It is considered good practice to only instances of the built-in `Error` object for user-defined errors in Promises. `Error` objects automatically store a stack trace, which can be used to debug an error by determining where it came from. If a Promise is rejected with a non-`Error` value, it can be difficult to determine where the rejection occurred.
+
+
+## Rule Details
+
+This rule aims to ensure that Promises are only rejected with `Error` objects.
+
+Examples of **incorrect** code for this rule:
+
+```js
+/*eslint no-reject-literal: "error"*/
+
+Promise.reject("something bad happened");
+
+Promise.reject(5);
+
+Promise.reject();
```

Comment:
The primary purpose of the `.reject`/`.catch` path is error handling. It is possible to use that path for control flow instead, but I don't think that's generally a good idea. For example, the synchronous equivalent is below. Its control flow unnecessarily convoluted, and could probably be refactored:

```js
try {
  // lots of code
  if (condition) {
    // jump to the error handler to start doing something else
    throw undefined;
  }
  // more code
} catch (err) {
  // do something else
}
```

I don't see why a pattern like that is any better when the control flow is asynchronous.